src/runtime-rs/crates/hypervisor/src/device/topology.rs

```diff
@@ -301,6 +403,20 @@ impl PCIeTopology {
         }
     }
 
+    pub fn find_vacant_port(&mut self, port_type: PCIePort) -> Option<TopologyPortDevice> {
```

Comment:
This function doesn't seem to be called either... are we missing a call to it from elsewhere in the PR?